ENH: Add django 1.9 compliant code

SantiagoTorres opened the pull request:
The management commands within django_pph, as well as the example application,
wasn't django 1.9 (or 1.10) ready, which made testing rather difficult. This PR
includes the code necessary to run these examples out of the bat with current
django versions.

SantiagoTorres committed:
FIX: example_project:settings: make 1.9 compliant

The settings.py file in the example project weren't django 1.9
compliant.

SantiagoTorres committed:
FIX: management:commands: make Django 1.9-ready

The management commands weren't Django 1.9 ready, as some of them still
used the *argv interface. Add the BaseCommand.parse_arguments()
interface to make the current implementation Django 1.9 (and 1.10)
ready.

lukpueh commented on example_project/example_project/settings.py:
```diff
@@ -53,34 +53,37 @@
 WSGI_APPLICATION = 'example_project.wsgi.application'
 
+# Internationalization
+# https://docs.djangoproject.com/en/1.6/topics/i18n/
```
Link `404`s. Why do you need an internationalization comment anyway? I don't see any related settings.

lukpueh replied:
Oops, sorry. Just saw, that the related lines are in https://github.com/PolyPasswordHasher/PolyPasswordHasher-Django/pull/9/files#diff-3684049e89acfc4131717dc0718de375R87.
Consider updating the link to
```
# https://docs.djangoproject.com/en/1.10/topics/i18n/
```
and move the comments down!

lukpueh commented on example_project/example_project/settings.py:
```diff
@@ -53,34 +53,37 @@
 
-# Database
-# https://docs.djangoproject.com/en/1.6/ref/settings/#databases
+DEBUG=True
```
Our code guide lines and django default settings (see below) propose a whitespace before and after the assign operator. Please use same style everywhere.

lukpueh commented on example_project/example_project/settings.py:
```diff
@@ -53,34 +53,37 @@
-PASSWORD_HASHERS = (
-    'django_pph.hashers.PolyPasswordHasherer',
+PASSWORD_HASHERS=(
+'django_pph.hashers.PolyPasswordHasher',
```
Please add 4 space indentation.

lukpueh reviewed (changes requested):
Tried with an example app (django first app tutorial) on djang 1.10.4. Ran management commands. Looked up idoms in recent django docs. The changes are legitimate and seem to work. Please change minor style nits and we can merge.

SantiagoTorres commented:
Thanks for the review! Glad to have a second pair of eyeballs on this code 👍

lukpueh commented:
Although this is not part of the PR I noticed two minor issues in the getting started readme.

(1) In the third verbatim block in [getting the sources](https://github.com/PolyPasswordHasher/PolyPasswordHasher-Django/wiki/Installing-PolyPasswordHasher-for-Django#getting-the-sources) one has to copy `django_pph` into the testapp dir using the command:
```shell
$ cp -r PolyPasswordHasher-Django-master/django_pph/ my_django_project/
```
On OS X the trailing slash after `django_pph` makes that it copies the contents of django_pph to my_django_project instead of the directory.

(2) In [Creating User Accounts](https://github.com/PolyPasswordHasher/PolyPasswordHasher-Django/wiki/Installing-PolyPasswordHasher-for-Django#for-new-password-databases-only-creating-user-accounts) the `createsuperuser` management command lists an optional `[username]` argument. Which it actually does not accept (at least not in django 1.10.4).

SantiagoTorres committed:
STY:example_proejct:settings:fix whitespace issues

The code was missing a couple of spaces before and after equal signs,
and some blocks were unindented. I also updated and moved some comments
around.

SantiagoTorres commented:
@lukpueh Updated the settings file. I'll also update the wiki soon.

lukpueh commented:
Great, IMHO ready to merge!
